sentry-core/src/units.rs

```diff
@@ -0,0 +1,260 @@
+//! Type definitions for Sentry metrics.
+
+use std::borrow::Cow;
+use std::fmt;
+
+/// The unit of measurement of a metric value.
+///
+/// Units augment metric values by giving them a magnitude and semantics. There are certain types of
+/// units that are subdivided in their precision:
+///  - [`DurationUnit`]: time durations
+///  - [`InformationUnit`]: sizes of information
+///  - [`FractionUnit`]: fractions
+///
+/// You are not restricted to these units, but can use any `&'static str` or `String` as a unit.
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Default)]
+pub enum MetricUnit {
+    /// A time duration, defaulting to `"millisecond"`.
+    Duration(DurationUnit),
+    /// Size of information derived from bytes, defaulting to `"byte"`.
+    Information(InformationUnit),
+    /// Fractions such as percentages, defaulting to `"ratio"`.
+    Fraction(FractionUnit),
+    /// user-defined units without builtin conversion or default.
+    Custom(Cow<'static, str>),
+    /// Untyped value without a unit (`""`).
+    #[default]
+    None,
+}
+
+impl MetricUnit {
+    /// Returns `true` if the metric_unit is [`None`].
+    pub fn is_none(&self) -> bool {
+        matches!(self, Self::None)
+    }
+}
+
+impl fmt::Display for MetricUnit {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            MetricUnit::Duration(u) => u.fmt(f),
+            MetricUnit::Information(u) => u.fmt(f),
+            MetricUnit::Fraction(u) => u.fmt(f),
+            MetricUnit::Custom(u) => u.fmt(f),
+            MetricUnit::None => f.write_str("none"),
+        }
+    }
+}
+
+impl std::str::FromStr for MetricUnit {
+    type Err = ParseMetricUnitError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(match s {
+            "nanosecond" | "ns" => Self::Duration(DurationUnit::NanoSecond),
+            "microsecond" => Self::Duration(DurationUnit::MicroSecond),
```

Comment:
Could do `μs` here, but I don't know if that's common.

Reply:
Good point, `μ` is an ASCII character so we could add this as an alias. We'd have to add this to the spec and Relay too, though.